streams/src/main/java/org/apache/kafka/streams/processor/internals/InitProcessorRecordContext.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.processor.internals;
+
+import org.apache.kafka.common.header.internals.RecordHeaders;
+import org.apache.kafka.common.utils.Time;
+
+import java.util.Objects;
+
+public class InitProcessorRecordContext extends ProcessorRecordContext {
+
+    private final Time time;
+
+    public InitProcessorRecordContext(final Time time) {
+        super(time.milliseconds(), -1, -1, null, new RecordHeaders());
+        this.time = time;
+    }
+
+    @Override
+    public long timestamp() {
+        return time.milliseconds();
+    }
+
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o) {
+            return true;
+        }
+        return o != null && getClass() == o.getClass();
+    }
+
+    @Override
+    @Deprecated
+    public int hashCode() {
```

Comment:
I don't think we should overwrite `hashCode()` at all, as the it's invalid to use (cf `ProcessorRecordContext.hashCode()`.